src/github.com/matrix-org/dendrite/clientapi/routing/createroom.go

```diff
@@ -170,7 +169,7 @@ func createRoom(req *http.Request, device *authtypes.Device,
 
 	var joinRules, historyVisibility string
 	switch r.Preset {
-	case presetPrivateChat:
+	case presetPrivateChat, "":
```

Comment:
I would probably have a `default` case to handle none specified, with a comment that we've already applied checks to ensure that we've checked valid values previously. As it is its non-obvious that `joinRules` and `historyVisibility` are always set to valid values